packages/welldone-wallet/src/lib/welldone.ts

```diff
@@ -295,6 +301,45 @@ const WelldoneWallet: WalletBehaviourFactory<InjectedWallet> = async ({
       };
     },
 
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async signMessage(message: SignMessageParams): Promise<SignedMessage> {
+      if (!_state.wallet) {
+        throw new Error("Wallet is not installed");
+      }
+
+      const account = await _getAccounts();
+      const accountId = account[0];
+
+      if (!accountId) {
+        throw new Error("Failed to find account for signing");
+      }
+
+      const serializedTx = serializeNep413(message as SignMessageParams);
+      const signed = await _state.wallet.request("near", {
+        method: "dapp:signMessage",
+        params: ["0x" + serializedTx.toString("hex")],
+      });
+
+      return (message as SignMessageParams).state
+        ? {
+            accountId,
+            publicKey: signed[0].publicKey,
+            signature: Buffer.from(
+              signed[0].signature.substr(2),
+              "hex"
+            ).toString("base64"),
+            state: (message as SignMessageParams).state,
+          }
+        : {
+            accountId,
+            publicKey: signed[0].publicKey,
+            signature: Buffer.from(
+              signed[0].signature.substr(2),
+              "hex"
+            ).toString("base64"),
+          };
+    },
```

Comment:
This suggestion is more of a personal preference but I think this code would read better if we avoid the ternary operator in this case:

```ts
const result = {
  accountId,
  publicKey: signed[0].publicKey,
  signature: Buffer.from(signed[0].signature.substr(2), "hex").toString(
    "base64"
  ),
};

if (message.state) {
  return { ...result, state: message.state };
}

return result;

```

Reply:
Thank you for the feedback.